File: src/Models/Book.php
<?php

namespace Models;

class Book
{
    protected int $id;
    protected string $title;
    protected string $author;
    protected int $year;
    protected string $language;
    protected string $description;

    public function __construct(int $id, string $author, string $title, string $description,
                                int $year, string $language)
    {
        $this->id = $id;
        $this->author = $author;
        $this->title = $title;
        $this->year = $year;
        $this->description = $description;
        $this->language = $language;
    }
}

File: src/main/java/com/example/demo/data/Book.java
package com.example.demo.data;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PUBLIC, force = true)
public class Book {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    private String title;
    private String description;
    private Integer publishingYear;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "author_id")
    private Author author;

    @Override
    public String toString() {
        return "Book{" +
                "id=" + id +
                ", title='" + title + '\'' +
                ", description='" + description + '\'' +
                ", publishingYear='" + publishingYear + '\'' +
                ", author=" + author.getFullName() +
                '}';
    }
}
